### PR TITLE
Enable Gosec linter

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
**What this PR does / why we need it**:
This linter checks for common insecure code patterns. Enabling this linter makes us less susceptible to known issues.

**Special notes for your reviewer**:
Review commits one by one to see the reasoning of each of the changes. I classified them by the vulnerability/issue they prevent.

Note that I ignored most issues that do not affect production code, and only affect tests or test helpers.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mhenriks** [comment]
very cool

/lgtm
/approve

**Comment by EduardGomezEscandell** [comment]
Rebased to see if SonarCloud stops acting out

File: hack/build/gosec.sh [hunk 1]
```diff
@@ -1,13 +0,0 @@
-#!/bin/bash
```
Comment (alromeros) [inline hack/build/gosec.sh:1]: There are still references to this file in the Makefile, is this intentional?

File: .golangci.yml [hunk 2]
```diff
@@ -24,10 +25,11 @@ issues:
   exclude:
     # Allow dot imports for test files
     - 'dot-imports: should not use dot import'
-    # Allow unused contexts
-    - "unused-parameter: parameter 'ctx' seems to be unused"
     # Allow unused variables (necessary sometimes to match interfaces)
     - "unused-parameter: parameter '.*' seems to be unused, consider removing or renaming it as _"
+    # Outdated warining, it only applies to Go < 1.22
```
Comment (alromeros) [inline .golangci.yml:30]: Typo: warining

**Review by alromeros — COMMENTED** [review]
Looks really good, thanks for the commit separation. Just a couple of comments.

File: hack/build/gosec.sh (hunk 1 above)
Comment (EduardGomezEscandell) [reply hack/build/gosec.sh:1]: No :sweat_smile: I got rid of it now

**Pull request merged**